File: mobile/src/main/java/com/windscribe/mobile/custom_view/ProgressFragment.java
/*
 * Copyright (c) 2021 Windscribe Limited.
 */

package com.windscribe.mobile.custom_view;

import android.os.Bundle;
import android.transition.Fade;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentTransaction;

import com.windscribe.mobile.R;

import butterknife.BindView;
import butterknife.ButterKnife;

public class ProgressFragment extends Fragment {

    @BindView(R.id.progressLabel)
    TextView mProgressLabel;

    private String progressText = "";

    public static ProgressFragment getInstance() {
        return new ProgressFragment();
    }

    public ProgressFragment() {

    }

    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
            Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.fragment_progress, container, false);
        ButterKnife.bind(this, view);
        return view;
    }

    @Override
    public void onViewCreated(@NonNull final View view, @Nullable final Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        updateProgressStatus(progressText);
    }

    @Override
    public void onDestroyView() {
        progressText = "";
        super.onDestroyView();
    }

    public void add(AppCompatActivity activity, int container, boolean addToBackStack) {
        setEnterTransition(new Fade().addTarget(R.id.progress_fragment_container));
        setExitTransition(new Fade().addTarget(R.id.progress_fragment_container));
        FragmentTransaction transaction = activity.getSupportFragmentManager()
                .beginTransaction()
                .add(container, this);
        if (addToBackStack) {
            transaction.addToBackStack(this.getClass().getName());
        }
        transaction.commit();
    }

    public void add(String progressText, AppCompatActivity activity, int container, boolean addToBackStack) {
        this.progressText = progressText;
        add(activity, container, addToBackStack);
    }

    public void finishProgress() {
        if (getActivity() != null) {
            getActivity().getSupportFragmentManager().popBackStack();
        }
    }

    public void updateProgressStatus(String call) {
        if (mProgressLabel != null) {
            progressText = call;
            mProgressLabel.setText(progressText);
        }
    }
}

File: tv/src/main/java/com/windscribe/tv/customview/ProgressFragment.kt
/*
 * Copyright (c) 2021 Windscribe Limited.
 */
package com.windscribe.tv.customview

import android.os.Bundle
import android.transition.Slide
import android.view.Gravity
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.Fragment
import com.windscribe.tv.R
import com.windscribe.tv.databinding.FragmentProgressBinding

class ProgressFragment : Fragment() {
    private var progressText = ""
    private lateinit var binding: FragmentProgressBinding
    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        binding = FragmentProgressBinding.inflate(inflater, container, false)
        return binding.root
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        updateProgressStatus(progressText)
    }

    override fun onDestroyView() {
        progressText = ""
        super.onDestroyView()
    }

    fun add(activity: AppCompatActivity, container: Int, addToBackStack: Boolean) {
        enterTransition = Slide(Gravity.BOTTOM)
            .addTarget(R.id.progress_fragment_container)
        val transaction = activity.supportFragmentManager
            .beginTransaction()
            .add(container, this)
        if (addToBackStack) {
            transaction.addToBackStack(this.javaClass.name)
        }
        transaction.commit()
    }

    fun add(
        progressText: String,
        activity: AppCompatActivity,
        container: Int,
        addToBackStack: Boolean
    ) {
        this.progressText = progressText
        add(activity, container, addToBackStack)
    }

    fun updateProgressStatus(call: String) {
        progressText = call
        binding.progressLabel.text = progressText
    }

    companion object {
        @JvmStatic
        val instance: ProgressFragment
            get() = ProgressFragment()
    }
}
